test/test.js

```diff
@@ -1073,31 +1080,134 @@ suite('GaiaHeader', function() {
-
-      obj[method] = function() {
-        clearTimeout(timeout);
-        obj[method] = real; // restore asap
-        var result = real.apply(obj, arguments);
-        resolve(result);
-        return result;
-      };
+  suite('accessibility', function() {
+    /**
+     * Accessibility test utils module tests the following things, amongst other
+     * checks (all at once).:
+     *  - ARIA attributes specific checks
+     *  - accesskey uniqueness if applicable
+     *  - Presence of alternative descriptions, labels and names
+     *  - Color contrast
+     *  - Markup is semantically correct from a11y standpoint
+     *  - Heading order
+     *  - Frame/document title and language
+     *  - Landmarks if applicable
+     *  - Keyboard focusability and tabindex
+     *
+     * Its checks are called at different stages and within different states of
+     * the component.
+     */
+
+    test('gaia-header passes accessibility checks with and without supported ' +
+      'action buttons', function(done) {
+      this.dom.innerHTML = `<gaia-header></gaia-header>
+        <gaia-header action="unsupported"></gaia-header>
+        <gaia-header action="menu"></gaia-header>
+        <gaia-header action="close"></gaia-header>
+        <gaia-header action="back"></gaia-header>`;
+
+      accessibility.check(this.dom).then(done, done);
+    });
+
+    test('gaia-header passes accessibility checks with its attributes ' +
+      'dynamically changing', function(done) {
+      this.dom.innerHTML = '<gaia-header action="back"><h1>Title</h1></gaia-header>';
+      var el = this.dom.firstElementChild;
+      var h1 = el.querySelector('h1');
+
+      accessibility.check(this.dom)
+        .then(() => {
+          /* change to another supported action */
+          el.setAttribute('action', 'close');
+          return accessibility.check(this.dom);
+        })
+        .then(() => {
+          /* change to an unsupported action */
+          el.setAttribute('action', 'unsupported');
+          return accessibility.check(this.dom);
+        })
+        .then(() => {
+          /* back to something supported via attribute field */
+          el.action = 'menu';
+          return accessibility.check(this.dom);
+        })
+        .then(() => {
+          /* set title start and title end */
+          el.setAttribute('title-start', '50');
+          el.setAttribute('title-end', '100');
+          return afterNext(el, 'runFontFit');
+        })
+        .then(() => accessibility.check(this.dom))
+        .then(() => {
+          /* invalid title end */
+          el.setAttribute('title-start', '50');
+          el.setAttribute('title-end', 'invalid');
+          return afterNext(el, 'runFontFit');
+        })
+        .then(() => accessibility.check(this.dom))
+        .then(() => {
+          /* title start only */
+          el.setAttribute('title-start', '50');
+          el.removeAttribute('title-end');
+          return afterNext(el, 'runFontFit');
```

Comment:
Maybe I'm missing something, but I don't understand how some of these tests are related to accessibility?